src/OpenSpaceToolkit/Astrodynamics/Trajectory/LocalOrbitalFrameFactory.cpp

```diff
@@ -50,18 +50,18 @@ struct SharedFrameEnabler : public Frame
     }
 };
 
-Shared<const Frame> LocalOrbitalFrameFactory::generateFrame(
-    const Instant& anInstant, const Vector3d& aPosition, const Vector3d& aVelocity
-) const
+Shared<const Frame> LocalOrbitalFrameFactory::generateFrame(const State& aState) const
 {
-    const String name = this->generateFrameName(anInstant, aPosition, aVelocity);
+    const State stateInParentFrame = aState.inFrame(parentFrameSPtr_);
```

Comment:
Nit: since we only need pos/vel and not all coordinate subsets might be portable to another frame without raising an error, we could reduce it first to avoid any issues

```suggestion
    const StateBuilder posVelStateBuilder = StateBuilder(state.getFrame(), {CartesianPosition::Default(), CartesianVelocity::Default()})
    const State posVelStateInParentFrame = posVelStateBuilder.reduce(aState).inFrame(parentFrameSPtr_);
```